### ISSUE TITLE
Remove PartialEq::ne implementation

### ISSUE THREAD

**Issue by dtolnay** [contain-rs/linked-hash-map#74]
The build is currently failing with this error:

```
error: re-implementing `PartialEq::ne` is unnecessary
   --> src/lib.rs:686:5
    |
686 |       fn ne(&self, other: &Self) -> bool {
    |  _____^ starting here...
687 | |         self.len() != other.len() || self.iter().ne(other)
688 | |     }
    | |_____^ ...ending here
    |
note: lint level defined here
   --> src/lib.rs:35:38
    |
35  | #![cfg_attr(feature = "clippy", deny(clippy))]
    |                                      ^^^^^^
    = help: for further information visit https://github.com/Manishearth/rust-clippy/wiki#partialeq_ne_impl
```

**Comment by apasel422** [comment]
I'd accept a PR.

**Comment by apasel422** [comment]
Fixed by #73.
